### PR TITLE
fix(connector): handle backwards compat parsing for `boolean` datatype

### PR DESCRIPTION
**Pull request by kwannoel**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

A `mysql` cdc-source may specify `mysql:boolean` to map to `rw:boolean`. In such case we have to handle it as [we did previously](https://github.com/risingwavelabs/risingwave/blob/b2b5a335a421667fd072add96617058aba52f361/src/connector/src/parser/mysql.rs#L62-L64), to avoid breaking this semantic.

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Review by wenym1 — COMMENTED** [review]
Can we cover this case in the e2e test? In `e2e_test/source_legacy/cdc/cdc.share_stream.slt`, when validating the result of `mysql_all_types`, we missed selecting `c_boolean` and `c_bit`.

**Review by wenym1 — APPROVED** [review]
LGTM!

File: e2e_test/source_legacy/cdc/cdc.share_stream.slt [hunk 1]
```diff
@@ -232,10 +232,10 @@ SELECT order_id,order_date,customer_name,product_id FROM orders_test order by or
 10003 2020-07-30 12:00:30 Edward 106
 
 query IIIIITTTTTTTTT
-SELECT c_tinyint, c_smallint, c_mediumint, c_integer, c_bigint, c_decimal, c_float, c_double, c_char_255, c_varchar_10000, c_date, c_time, c_datetime, c_timestamp FROM mysql_all_types order by c_bigint;
+SELECT c_boolean, c_bit, c_tinyint, c_smallint, c_mediumint, c_integer, c_bigint, c_decimal, c_float, c_double, c_char_255, c_varchar_10000, c_date, c_time, c_datetime, c_timestamp FROM mysql_all_types order by c_bigint;
 ----
--128 -32767 -8388608 -2147483647 -9223372036854775807 -10 -10000 -10000 a b 1001-01-01 00:00:00 1998-01-01 00:00:00 1970-01-01 00:00:01+00:00
-NULL NULL -8388608 -2147483647 9223372036854775806 -10 -10000 -10000 c d 1001-01-01 NULL 2000-01-01 00:00:00 NULL
+t, t, -128 -32767 -8388608 -2147483647 -9223372036854775807 -10 -10000 -10000 a b 1001-01-01 00:00:00 1998-01-01 00:00:00 1970-01-01 00:00:01+00:00
```
Comment (wenym1) [inline e2e_test/source_legacy/cdc/cdc.share_stream.slt:237]: Seems that we shouldn't use comma to separate the values.

**Review by chenzl25 — APPROVED** [review]
LGTM

**Pull request merged**